### PR TITLE
File linking

### PR DESCRIPTION
**Pull request by PiotrZakrzewski**
- RESTfull endpoint for adding new links to file collections
- ddl for postgres
- ddl for oracle
- test data

### CODE REVIEW

File: transmart-core-db/grails-app/domain/org/transmartproject/db/storage/LinkedFileCollection.groovy [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    String uuid // uniformly unique identifier generated by the source system
+    StorageSystem sourceSystem // origin of the collection
+    String name  // name of the collection, or name the file for single file collection systems (e.g. MongoDB)
+
+    static belongsTo = [
+            sourceSystem:        StorageSystem,
+    ]
+
+    static mapping = {
+        table          schema:   'I2B2DEMODATA'
```
Comment (gijskant) [inline transmart-core-db/grails-app/domain/org/transmartproject/db/storage/LinkedFileCollection.groovy]: I'm not sure if `i2b2demodata` is the correct schema for this. Let's discuss.

File: transmart-core-db/grails-app/domain/org/transmartproject/db/storage/LinkedFileCollection.groovy [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package org.transmartproject.db.storage
+
+/**
+ * Created by piotrzakrzewski on 02/12/2016.
+ */
+class LinkedFileCollection {
+
+    String study // Id of the Study with which the collection is associated
```
Comment (gijskant) [inline transmart-core-db/grails-app/domain/org/transmartproject/db/storage/LinkedFileCollection.groovy]: Is there a reason why not link to `org.transmartproject.db.i2b2data.Study`?

File: transmart-rest-api/grails-app/controllers/org/transmartproject/rest/StorageController.groovy [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    @Autowired
+    UsersResource usersResource
+
+    StorageController() {
+        super(LinkedFileCollection)
+    }
+
+    @Override
+    def show() {
+        def fileCollection = queryForResource(params.id)
```
Comment (gijskant) [inline transmart-rest-api/grails-app/controllers/org/transmartproject/rest/StorageController.groovy]: Or add is as parameter to the function (`def show(Long id)` or something like that). But I'm never sure which method is best in Grails, so choose your own preferred way.

File: transmart-rest-api/src/integration-test/groovy/org/transmartproject/rest/StorageControllerSpec.groovy [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        def bodyContent = ['name'        : 'new file Link',
+                           'sourceSystem': 1,
+                           'study'       : 'STUDY_ID_2',
+                           'uuid'        : 'aaaaa-bbbbb-ccccccccccccccc'] as JSON
+        def response = post "/$VERSION/files", {
+            contentType "application/json"
+            json bodyContent
+        }
+        then:
+        response.status == 201
```
Comment (gijskant) [inline transmart-rest-api/src/integration-test/groovy/org/transmartproject/rest/StorageControllerSpec.groovy]: I would expect there to be a fetch as well, to make sure the `post` was really succesful.

File: transmart-rest-api/grails-app/controllers/RestApiUrlMappings.groovy [hunk 5]
```diff
@@ -65,6 +65,36 @@ class RestApiUrlMappings {
             "/tree_nodes"(method: 'GET', controller: 'tree', action: 'index') {
                 apiVersion = 'v2'
             }
+            "/files"(method: 'GET', controller: 'storage', action: 'index') {
+                apiVersion = "v2"
+            }
+            "/files/$id"(method: 'GET', controller: 'storage', action: 'show') {
+                apiVersion = "v2"
+            }
+            "/files"(method: 'POST', controller: 'storage', action: 'save') {
```
Comment (gijskant) [inline transmart-rest-api/grails-app/controllers/RestApiUrlMappings.groovy]: To do: add access policy on save, update, delete.

Comment (PiotrZakrzewski) [inline transmart-rest-api/grails-app/controllers/RestApiUrlMappings.groovy]: done

File: transmart-rest-api/grails-app/controllers/org/transmartproject/rest/StorageController.groovy [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+        currentUser.checkAccess(READ, fileCollection.study)
+        respond fileCollection
+    }
+
+    @Override
+    def save() {
+        User user = (User) usersResource.getUserFromUsername(currentUser.username)
+        if (!user.admin) {
+            throw new AccessDeniedException("Listing all Linked File Collections " +
```
Comment (gijskant) [inline transmart-rest-api/grails-app/controllers/org/transmartproject/rest/StorageController.groovy]: Copy pasted message?

File: transmart-rest-api/grails-app/controllers/org/transmartproject/rest/StorageSystemController.groovy [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+import org.transmartproject.core.exceptions.AccessDeniedException
+import org.transmartproject.core.users.UsersResource
+import org.transmartproject.db.storage.StorageSystem
+import org.transmartproject.db.user.User
+import org.transmartproject.rest.misc.CurrentUser
+
+/**
+ * Created by piotrzakrzewski on 09/12/2016.
+ */
+class StorageSystemController extends RestfulController {
```
Comment (gijskant) [inline transmart-rest-api/grails-app/controllers/org/transmartproject/rest/StorageSystemController.groovy]: I think the `update` function is missing.

Comment (PiotrZakrzewski) [inline transmart-rest-api/grails-app/controllers/org/transmartproject/rest/StorageSystemController.groovy]: added

**Pull request merged**